cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -117,14 +117,86 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
      * ```
      */
     private fun handleWithStatement(node: Python.AST.With): Block {
+        /** Prepares the __exit__(manager, None, None, None) call for the else-block. */
+        fun generateExitCallWithNone(
+            manager: Expression,
+            withItem: Python.AST.withitem
+        ): Statement {
+            val exitCallWithNone =
+                newMemberCallExpression(
+                        callee = newMemberExpression(name = "__exit__", base = manager).implicit(),
+                        rawNode = withItem
+                    )
+                    .implicit()
+            exitCallWithNone.addArgument(newLiteral(null).implicit())
+            exitCallWithNone.addArgument(newLiteral(null).implicit())
+            exitCallWithNone.addArgument(newLiteral(null).implicit())
+            return exitCallWithNone
+        }
+
+        /**
+         * Prepares the if-statement which is the body of the catch block. This includes the call of
+         * `__exit__(manager, *sys.exc_info())`, the negation and the throw statement.
+         */
+        fun generateExitCallWithSysinfo(
+            manager: Expression,
+            withItem: Python.AST.withitem
+        ): Statement {
+            val exitCallWithSysExec =
+                newMemberCallExpression(
+                        callee = newMemberExpression(name = "__exit__", base = manager).implicit(),
+                        rawNode = withItem
+                    )
+                    .implicit()
+            val starOp = newUnaryOperator("*", false, false)
+            starOp.input =
+                newMemberExpression("exec_info", newReference("sys").implicit()).implicit()
+            exitCallWithSysExec.addArgument(starOp)
+
+            val ifStmt = newIfStatement().implicit()
+            // TODO: Needs #1733 or 1741, then add:
+            //   ifStmt.thenStatement = newThrowStatement().implicit()
```

Comment:
would it make sense to add a unary operator call to "throw" in the mean time (as its modelled for C++?) or should we just hope that we don't forget it here? Maybe add an issue?

Reply:
I'd prefer to leave a comment in the respective PRs since they should be about to be finalized too

Reply:
ok